pageserver/src/bin/pageserver.rs

```diff
@@ -340,8 +332,6 @@ fn start_pageserver(conf: &'static PageServerConf) -> anyhow::Result<()> {
         },
     );
 
-    set_build_info_metric(GIT_VERSION);
```

Comment:
I think that should be around here still, since otherwise with the current code, we risk being scrapped by prometheus and then bail due to some occupied port.
While not critical, still quite odd and racy.

Reply:
Can you elaborate? What do you mean by being scrapped by prometheus?

Reply:
The way metrics work: we expose the `/metrics` HTTP endpoint, prometheus scraps those periodically.

In our case, when we store the version metric and manage to get past `let http_listener = tcp_listener::bind(http_addr)?;`, we expose this version metric value for scrapping.

Yet, there's a number of further `?` to be able to fail on before we actually can consider the pageserver as "successfully started", e.g.
```rust
let pg_addr = &conf.listen_pg_addr;
info!("Starting pageserver pg protocol handler on {pg_addr}");
let pageserver_listener = tcp_listener::bind(pg_addr)?;
```

The metric was created to show which version is currently deployed, does not make much sense to spoil it with fake values (even shortly)